src/AtomicLaunch.sol

```diff
@@ -0,0 +1,139 @@
+    constructor(address _governance) {
+        governance = _governance;
+    }
+
+    /***************************************
+                    MODIFIER
+    ****************************************/
+
+    modifier onlyGovernance() {
+        require(msg.sender == governance, "not-governance!");
+        _;
+    }
+
+    /***************************************
+               ADMIN - GOVERNANCE
+    ****************************************/
+
+    function addOracle(address _oracleAddress) external onlyGovernance {
+        require(_oracleAddress != address(0), "zero-address!");
+        _oracles.add(_oracleAddress);
+    }
+
+    function launch(uint256 citadelToLiquidty, uint256 wbtcToLiquidity)
+        external
+        onlyGovernance
+    {
+        uint256 initialPoolPrice = _initialPoolPrice();
+
+        address[2] memory coins;
+        coins[0] = CITADEL;
+        coins[1] = WBTC;
+
+        address poolAddress = ICurvePoolFactory(CURVE_POOL_FACTORY).deploy_pool(
+            "CTDL/wBTC",
+            "CTDL",
+            coins,
+            400000,
+            145000000000000,
+            26000000,
+            45000000,
+            2000000000000,
+            230000000000000,
+            146000000000000,
+            5000000000,
+            600,
+            initialPoolPrice
+        );
+
+        ICurvePool pool = ICurvePool(poolAddress);
+
+        require(
+            IERC20(CITADEL).balanceOf(address(this)) >= citadelToLiquidty,
+            "<citadelToLiquidty!"
+        );
+        require(
+            IERC20(WBTC).balanceOf(address(this)) >= wbtcToLiquidity,
+            "<wbtcToLiquidity!"
+        );
+
+        IERC20(CITADEL).approve(poolAddress, citadelToLiquidty);
+        IERC20(WBTC).approve(poolAddress, wbtcToLiquidity);
+
+        uint256[2] memory amounts;
+        amounts[0] = citadelToLiquidty;
+        amounts[1] = wbtcToLiquidity;
+
+        pool.add_liquidity(amounts, 0);
+
+        require(
+            pool.balances(0) >= citadelToLiquidty,
+            "<pool-curve-citadelToLiquidty!"
+        );
+        require(
+            pool.balances(1) >= wbtcToLiquidity,
+            "<pool-curve-wbtcToLiquidity!"
+        );
+
+        _setCurvePoolInOracles(poolAddress);
+
+        // NOTE: mainly for clean event, so no needs for mega-detective work
+        emit PoolCreated(poolAddress);
+    }
+
+    function _initialPoolPrice() internal returns (uint256 _poolPrice) {
+        (uint256 wbtcOraclePricing, , bool valid) = safeLatestAnswer(
+            wbtcBtcPriceFeed
+        );
+
+        require(valid, "not-valid!");
+
+        uint256 wbtcPrice = wbtcOraclePricing / 10**wbtcBtcPriceFeed.decimals();
+
+        _poolPrice = (wbtcPrice / CITADEL_PRICE) * 1e18;
+    }
+
+    function _setCurvePoolInOracles(address _ctdlWbtcCurvePool) internal {
+        for (uint256 i = 0; i < _oracles.length(); i++) {
+            IMedianOracleProvider(_oracles.at(i)).setCurvePool(
```

Comment:
Where is `setCurvePool` implemented?

Reply:
The premise here was to allow the setting of the curve pool after creation, so the contracts can be deployed before this process. Due to the benefits of immutables, I suggest moving to a Factory pattern where this contract deploys the oracles itself.

Reply:
At the end in the latest commits we decided for the contract itself to deploy the oracles using the curve pool address directly. So we did not end up adding this new method